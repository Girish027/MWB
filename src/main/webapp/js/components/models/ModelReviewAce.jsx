import React, { Component } from 'react';
import AceEditor from 'react-ace';
import PropTypes from 'prop-types';
import 'brace/mode/json';
import 'brace/theme/textmate';
import 'brace/theme/kuroir';

class ModelReviewAce extends Component {
  constructor(props) {
    super(props);
    this.props = props;
  }

  render() {
    const {
      name, onLoad, onChange, value, height, width, readOnly, onBlur,
    } = this.props;

    let theme = 'kuroir';
    if (!readOnly) {
      theme = 'textmate';
    }

    return (
      <AceEditor
        mode="json"
        theme={theme}
        name={name}
        onLoad={onLoad}
        onChange={onChange}
        onBlur={onBlur}
        fontSize={14}
        showPrintMargin={false}
        showGutter
        highlightActiveLine
        readOnly={readOnly}
        value={value}
        height={`${height}px`}
        width={`${width}px`}
        wrapEnabled
        setOptions={{
          firstLineNumber: 1,
          enableBasicAutocompletion: false,
          enableLiveAutocompletion: false,
          enableSnippets: false,
          showLineNumbers: true,
          tabSize: 2,
        }}
      />
    );
  }
}

ModelReviewAce.propTypes = {
  name: PropTypes.string,
  onLoad: PropTypes.func,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  value: PropTypes.string,
  height: PropTypes.number,
  width: PropTypes.number,
  readOnly: PropTypes.bool,
};

export default ModelReviewAce;
