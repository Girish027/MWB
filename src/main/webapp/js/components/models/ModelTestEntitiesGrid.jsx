import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Table } from '@tfs/ui-components';
import Constants from 'constants/Constants';

export default class ModelTestEntitiesGrid extends Component {
  constructor(props) {
    super(props);
    this.props = props;
  }

  get columns() {
    const columns = [
      {
        Header: Constants.MODEL_TEST_ENTITIES_TABLE.name.header,
        accessor: Constants.MODEL_TEST_ENTITIES_TABLE.name.id,
        id: Constants.MODEL_TEST_ENTITIES_TABLE.name.header,
      },
      {
        Header: Constants.MODEL_TEST_ENTITIES_TABLE.value.header,
        accessor: Constants.MODEL_TEST_ENTITIES_TABLE.value.id,
      },
    ];
    return columns;
  }

  render() {
    return (
      <Table
        id="ModelTestEntitiesGrid"
        resizable
        data={this.props.data}
        columns={this.columns}
        showPagination={false}
        defaultSorted={[
          {
            id: Constants.MODEL_TEST_ENTITIES_TABLE.name.header,
            desc: false,
          },
        ]}
      />
    );
  }
}

ModelTestEntitiesGrid.propTypes = {
  data: PropTypes.array.isRequired,
};
