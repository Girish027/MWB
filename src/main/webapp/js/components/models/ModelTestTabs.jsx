import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { connect } from 'react-redux';
import * as ramda from 'ramda';
import URLSearchParams from '@ungap/url-search-params';
import { RouteNames, urlMap } from 'utils/routeHelpers';
import { changeRoute, displayWarningRequestMessage } from 'state/actions/actions_app';
import * as actionsTagDatasets from 'state/actions/actions_tag_datasets';
import { getDataUrl, pathKey } from 'utils/apiUrls';
import {
  tuneSelectedModel,
  clearModelTestResults,
  listBatchTests,
  testModel,
  testSpeechModel,
} from 'state/actions/actions_models';
import * as headerActions from 'state/actions/actions_header';
import Constants from 'constants/Constants';
import Model from 'model';
import BatchTestContainer from 'components/models/batchTest/BatchTestContainer';
import UtteranceTestContainer from 'components/models/UtteranceTestContainer';
import {
  LegacyGrid,
  LegacyRow,
  Pencil,
  Tabs,
} from '@tfs/ui-components';


export class ModelTestTabs extends Component {
  constructor(props) {
    super(props);
    this.onReviewDatasetsClick = this.onReviewDatasetsClick.bind(this);
    this.onTuneModelClick = this.onTuneModelClick.bind(this);
    this.onClickBatchTestTab = this.onClickBatchTestTab.bind(this);
    this.onRunSingleUtterance = this.onRunSingleUtterance.bind(this);
    this.clearRunSingleUtterance = this.clearRunSingleUtterance.bind(this);
    this.getTabs = this.getTabs.bind(this);

    this.tabs = Constants.TEST_TABS;
    this.chunks = [];
    this.routes = {
      [this.tabs.utteranceTestTab]: RouteNames.TESTMODEL,
      [this.tabs.batchTestTab]: RouteNames.BATCHTESTMODEL,
    };

    this.actionItemStyle = {
      fontWeight: 'normal',
      ':focus': {
        outline: 'none',
      },
    };

    this.state = {
      runSingleUtterance: '',
    };
  }

  componentDidMount() {
    const {
      dispatch, model, projectId, client, utterance,
    } = this.props;

    const dataCommon = {
      projectId,
      modelId: model.modelToken,
      clientId: client.id,
    };
    const modelType = model ? model.modelType : '';
    const audioFile = getDataUrl(pathKey.dummySpeechTest);

    // this.chunks.push();
    // const recordedData = new Blob(this.chunks, { type: Constants.AUDIO_TYPE_WAV });
    // const reader = new FileReader();
    // reader.readAsBinaryString(recordedData);

    const digitalData = {
      ...dataCommon,
      testModelType: Constants.DIGITAL_MODEL,
      utterances: [
        utterance,
      ],
    };

    const speechData = {
      ...dataCommon,
      fileType: Constants.FILE_TYPE.UPLOADED,
      audioFile,
    };

    const actionItems = [{
      type: 'flat',
      label: Constants.TUNE_VERSION,
      icon: Pencil,
      onClick: this.onTuneModelClick,
      styleOverride: this.actionItemStyle,
    }];
    if (modelType === Constants.DIGITAL_MODEL) {
      dispatch(testModel(digitalData, false));
    } else if (modelType === Constants.DIGITAL_SPEECH_MODEL) {
      // const reader = new FileReader();
      // reader.onload = () => {
      //  dispatch(testSpeechModel(speechData, true));
      // };
      // reader.readAsBinaryString(audioFile);
    }
    dispatch(clearModelTestResults());
    dispatch(headerActions.setActionItems(actionItems));
  }

  componentWillUnmount() {
    const { dispatch } = this.props;
    dispatch(headerActions.setActionItems([]));
  }

  onTuneModelClick() {
    const {
      modelId, client, history, projectId, dispatch, model,
    } = this.props;
    dispatch(tuneSelectedModel(model));
    dispatch(changeRoute(RouteNames.TUNEMODEL, { client, projectId, modelId }, history));
  }

  onReviewDatasetsClick() {
    const {
      projectId, history, client, dispatch,
    } = this.props;
    dispatch(actionsTagDatasets.setIncomingFilter({ projectId, datasets: [] }));
    dispatch(changeRoute(RouteNames.TAG_DATASETS, { client, projectId }, history));
  }

  onClickBatchTestTab() {
    const {
      dispatch, model, client, projectId, userId, csrfToken,
    } = this.props;
    if (_.isNil(model) || _.isEmpty(model)) {
      dispatch(displayWarningRequestMessage(Constants.UNKNOWN_MODEL));
      return;
    }
    const data = {
      userId,
      csrfToken,
      projectId,
      modelId: model.modelToken,
      clientId: client.id,
    };
    dispatch(listBatchTests(data));
  }

  onRunSingleUtterance(transcription) {
    this.setState({
      runSingleUtterance: transcription,
    });
  }

  clearRunSingleUtterance() {
    this.setState({
      runSingleUtterance: '',
    });
  }

  onTabSelected = (selectedTab, selectedTabIndex) => {
    const {
      client, projectId, modelId, history, dispatch,
    } = this.props;
    const { batchTestTab, utteranceTestTab } = this.tabs;
    let tab = '';
    switch (selectedTabIndex) {
    case 0:
      tab = utteranceTestTab;
      break;
    case 1:
      tab = batchTestTab;
      this.onClickBatchTestTab();
      break;
    default:
      tab = utteranceTestTab;
      break;
    }
    dispatch(changeRoute(this.routes[tab], {
      client, projectId, modelId,
    }, history));
  }

  getTabs() {
    const { runSingleUtterance = '' } = this.state;
    const { batchTestTab, utteranceTestTab } = this.tabs;
    const {
      dispatch,
      match,
      utterance,
      model,
    } = this.props;

    let selectedTabIndex = 0;
    if (match.path === urlMap.BATCHTESTMODEL) {
      selectedTabIndex = 1;
    }

    const tabPanels = [(
      <UtteranceTestContainer
        key={selectedTabIndex}
        {...this.props}
        utterance={utterance}
        runSingleUtterance={runSingleUtterance}
        clearRunSingleUtterance={this.clearRunSingleUtterance}
      />
    ), (
      <BatchTestContainer
        key={selectedTabIndex}
        {...this.props}
        dispatch={dispatch}
        onTuneModelClick={this.onTuneModelClick}
        onReviewDatasetsClick={this.onReviewDatasetsClick}
        onRunSingleUtterance={this.onRunSingleUtterance}
      />
    )];
    let tabsTitle = [utteranceTestTab, batchTestTab];
    if (model.modelType === Constants.SPEECH_MODEL) {
      tabsTitle = [utteranceTestTab];
    }
    return (
      <div id="ReadProject">
        <Tabs
          tabs={tabsTitle}
          tabPanels={tabPanels}
          onTabSelected={this.onTabSelected}
          align="left"
          selectedIndex={selectedTabIndex}
        />
      </div>
    );
  }


  render() {
    return (
      <div id="ModelTest">
        <LegacyGrid fluid>
          <LegacyRow>
            {this.getTabs()}
          </LegacyRow>
        </LegacyGrid>
      </div>
    );
  }
}

ModelTestTabs.propTypes = {
  userId: PropTypes.string,
  csrfToken: PropTypes.string,
  app: PropTypes.object,
  digitalModelTestResults: PropTypes.object,
  projectId: PropTypes.string,
  speechModelTestResults: PropTypes.object,
  modelId: PropTypes.string,
  client: PropTypes.object,
  models: PropTypes.object,
  project: PropTypes.object,
  model: PropTypes.object,
  datasets: PropTypes.object,
};

ModelTestTabs.defaultProps = {
  digitalModelTestResults: {},
  speechModelTestResults: {},
};

const mapStateToProps = (state, ownProps) => {
  // TODO: URI Utils.js
  const searchString = ramda.path(['location', 'search'], ownProps) || '';
  const query = new URLSearchParams(searchString);

  const projectId = state.projectListSidebar.selectedProjectId;
  const modelId = query.get('modelid');

  const { userId, csrfToken } = state.app;
  const { searchResults = [] } = state.taggingGuide;
  const utterance = searchResults.length > 0 ? searchResults[0].examples : 'make a reservation';
  // some of the props are used in the child components.
  // TODO: more optimization for the whole state.
  return {
    userId,
    csrfToken,
    app: state.app,
    digitalModelTestResults: state.projectsManager.digitalModelTestResults,
    speechModelTestResults: state.projectsManager.speechModelTestResults,
    projectId,
    modelId,
    utterance,
    client: state.header.client,
    models: Model.ProjectsManager.getModelsByProjectId(projectId) || null,
    project: Model.ProjectsManager.getProject(projectId) || null,
    model: Model.ProjectsManager.getModel(projectId, modelId) || {},
    datasets: Model.ProjectsManager.getDatasetsByProjectId(projectId, true) || null,
  };
};

const mapDispatchToProps = dispatch => ({
  dispatch,
});

export default connect(mapStateToProps, mapDispatchToProps)(ModelTestTabs);
