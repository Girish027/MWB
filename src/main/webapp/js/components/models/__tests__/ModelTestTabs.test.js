import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import store from 'state/configureStore';
import * as actionsTagDatasets from 'state/actions/actions_tag_datasets';
import * as actionsModels from 'state/actions/actions_models';
import * as actionsHeader from 'state/actions/actions_header';
import * as actionsApp from 'state/actions/actions_app';
import { RouteNames } from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import {
  Pencil,
} from '@tfs/ui-components';
import { ModelTestTabs } from 'components/models/ModelTestTabs';
import ConnectedModelTestTabs from 'components/models/ModelTestTabs';

describe('<ModelTestTabs />', () => {
  const match = {
    params: {
      projectId: '7',
      modelId: '3',
    },
  };

  const modelTestResults = {
    projectId: '7',
    modelId: 'b61693f6-fa65-4308-af09-b32872dfb5f6',
    type: 'UTTERANCES',
    status: 'SUCCESS',
    evaluations: [{
      utterance: 'i want to cancel for june 13',
      intents: [{
        intent: 'reservation-query',
        score: 0.49544276870123316,
      }, {
        intent: 'agent-query',
        score: 0.42004871402243316,
      }, {
        intent: 'None_None',
        score: 0.14285217997731586,
      }],
      transformations: [{
        id: 'rx-mapper.non-breaking-space-regex',
        result: 'i want to cancel for june 13',
      }, {
        id: 'wsp',
        result: 'i want to cancel for june 13',
      }],
      entities: [{
        name: 'date',
        value: 'june 13',
      }],
    }],
  };

  const props = {
    userId: 'abc',
    csrfToken: '123',
    modelTestResults,
    projectId: '7',
    modelId: '3',
    client: {
      id: '2',
    },
    match: {
      path: '/models/test',
    },
    model: {
      modelToken: 'b61693f6-fa65-4308-af09-b32872dfb5f6',
    },
    dispatch: () => {},
    history: {},
  };

  let wrapper;

  afterAll(() => {
    jest.clearAllMocks();
  });


  describe('Creating an instance', () => {
    beforeAll(() => {
      wrapper = mount(<Provider store={store}>
        <MemoryRouter
          initialEntries={['/']}
          initialIndex={0}
        >
          <ConnectedModelTestTabs
            match={match}
            location={{
              search: '?clientid=247ai&appid=aisha&projectid=7&modelid=3',
            }}
          />
        </MemoryRouter>
      </Provider>);
    });

    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });
  });


  describe('Snapshots', () => {
    test('renders correctly with two tabs', () => {
      wrapper = shallow(<ModelTestTabs
        {...props}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when a utterance tab currently active', () => {
      wrapper = shallow(<ModelTestTabs
        {...props}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
    test('renders correctly when a batch test tab currently active', () => {
      const batchtestprops = Object.assign({}, props, { match: { path: '/models/batchtest' } });
      wrapper = shallow(<ModelTestTabs
        {...batchtestprops}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeAll(() => {
      props.dispatch = jest.fn();

      actionsModels.tuneSelectedModel = jest.fn(() => 'called tuneSelectedModel');
      actionsModels.clearModelTestResults = jest.fn(() => 'called clearModelTestResults');
      actionsModels.testModel = jest.fn(() => 'called testModel');
      actionsModels.listBatchTests = jest.fn(() => 'called listBatchTests');

      actionsHeader.setActionItems = jest.fn(() => 'called setActionItems');

      actionsApp.changeRoute = jest.fn(() => 'called changeRoute');
      actionsApp.displayWarningRequestMessage = jest.fn(() => 'called displayWarningRequestMessage');

      actionsTagDatasets.setIncomingFilter = jest.fn(() => 'called setIncomingFilter');
    });

    beforeEach(() => {
      wrapper = shallow(<ModelTestTabs
        {...props}
      />);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    describe('componentDidMount:', () => {
      test('should dispatch action to clear test results', () => {
        wrapper.instance().componentDidMount();
        expect(actionsModels.clearModelTestResults).toHaveBeenCalled;
        expect(props.dispatch).toHaveBeenCalledWith('called clearModelTestResults');
      });

      test('should dispatch action toset action items in page header', () => {
        wrapper.instance().componentDidMount();
        expect(actionsHeader.setActionItems).toHaveBeenCalledWith([{
          type: 'flat',
          label: Constants.TUNE_VERSION,
          icon: Pencil,
          onClick: wrapper.instance().onTuneModelClick,
          styleOverride: expect.any(Object),
        }]);
        expect(props.dispatch).toHaveBeenCalledWith('called setActionItems');
      });
    });

    describe('componentWillUnmount:', () => {
      test('should dispatch action to remove action items in page header', () => {
        wrapper.instance().componentWillUnmount();
        expect(actionsHeader.setActionItems).toHaveBeenCalledWith([]);
        expect(props.dispatch).toHaveBeenCalledWith('called setActionItems');
      });
    });

    describe('onTabSelected:', () => {
      test('should dispatch action to change the route to testmodel', () => {
        const {
          client, history, projectId, modelId,
        } = props;
        wrapper.instance().onTabSelected(Constants.TEST_TABS.utteranceTestTab, 0);
        expect(actionsApp.changeRoute)
          .toHaveBeenCalledWith(RouteNames.TESTMODEL, { client, projectId, modelId }, history);
        expect(props.dispatch).toHaveBeenCalledWith('called changeRoute');
      });

      test('should dispatch action to change the route to batchtest', () => {
        const {
          client, history, projectId, modelId,
        } = props;
        wrapper.instance().onTabSelected(Constants.TEST_TABS.batchTestTab, 1);
        expect(actionsApp.changeRoute)
          .toHaveBeenCalledWith(RouteNames.BATCHTESTMODEL, { client, projectId, modelId }, history);
        expect(props.dispatch).toHaveBeenCalledWith('called changeRoute');
      });

      test('should dispatch action to change the route to testmodel', () => {
        const {
          client, history, projectId, modelId,
        } = props;
        wrapper.instance().onTabSelected(Constants.TEST_TABS.batchTestTab, 2);
        expect(actionsApp.changeRoute)
          .toHaveBeenCalledWith(RouteNames.TESTMODEL, { client, projectId, modelId }, history);
        expect(props.dispatch).toHaveBeenCalledWith('called changeRoute');
      });
    });

    describe('onTuneModelClick:', () => {
      test('should dispatch action to tune model', () => {
        wrapper.instance().onTuneModelClick();
        expect(actionsModels.tuneSelectedModel).toHaveBeenCalledWith(props.model);
        expect(props.dispatch).toHaveBeenCalledWith('called tuneSelectedModel');
      });

      test('should dispatch action to change route to tune model view', () => {
        const {
          modelId, client, history, projectId,
        } = props;
        wrapper.instance().onTuneModelClick();
        expect(actionsApp.changeRoute)
          .toHaveBeenCalledWith(RouteNames.TUNEMODEL, { client, projectId, modelId }, history);
        expect(props.dispatch).toHaveBeenCalledWith('called changeRoute');
      });
    });

    describe('onReviewDatasetsClick:', () => {
      test('should dispatch action to set filter on Tag Datasets view', () => {
        const { projectId } = props;
        wrapper.instance().onReviewDatasetsClick();
        expect(actionsTagDatasets.setIncomingFilter).toHaveBeenCalledWith({ projectId, datasets: [] });
        expect(props.dispatch).toHaveBeenCalledWith('called setIncomingFilter');
      });

      test('should dispatch action to change route to tag datasets view', () => {
        const {
          client, history, projectId,
        } = props;
        wrapper.instance().onReviewDatasetsClick();
        expect(actionsApp.changeRoute)
          .toHaveBeenCalledWith(RouteNames.TAG_DATASETS, { client, projectId }, history);
        expect(props.dispatch).toHaveBeenCalledWith('called changeRoute');
      });
    });

    describe('onClickBatchTestTab:', () => {
      test('should dispatch action to display warning message if model is not present', () => {
        wrapper = shallow(<ModelTestTabs
          {...props}
          model={{}}
        />);
        wrapper.instance().onClickBatchTestTab();
        expect(actionsApp.displayWarningRequestMessage).toHaveBeenCalledWith(Constants.UNKNOWN_MODEL);
        expect(props.dispatch).toHaveBeenCalledWith('called displayWarningRequestMessage');
      });

      test('should not dispatch action to list batch tests, if model is not available', () => {
        wrapper = shallow(<ModelTestTabs
          {...props}
          model={{}}
        />);
        wrapper.instance().onClickBatchTestTab();
        expect(actionsModels.listBatchTests).not.toHaveBeenCalled;
        expect(props.dispatch).not.toHaveBeenCalledWith('called listBatchTests');
      });

      test('should dispatch action to list existing batch tests', () => {
        const {
          model, client, projectId, userId, csrfToken,
        } = props;
        const expectedData = {
          userId,
          csrfToken,
          projectId,
          modelId: model.modelToken,
          clientId: client.id,
        };
        wrapper.instance().onClickBatchTestTab();
        expect(actionsModels.listBatchTests).toHaveBeenCalledWith(expectedData);
        expect(props.dispatch).toHaveBeenCalledWith('called listBatchTests');
      });
    });

    describe('onRunSingleUtterance:', () => {
      test('should set the active tab to utterance and update incoming utterance', () => {
        wrapper.setState({
          runSingleUtterance: '',
        });
        wrapper.instance().onRunSingleUtterance('how are you');
        expect(wrapper.state().runSingleUtterance).toEqual('how are you');
      });
    });

    describe('clearRunSingleUtterance:', () => {
      test('should clear the state for incoming utterance', () => {
        wrapper.setState({
          runSingleUtterance: 'hi...',
        });
        wrapper.instance().clearRunSingleUtterance();
        expect(wrapper.state().runSingleUtterance).toEqual('');
      });
    });
  });
});
