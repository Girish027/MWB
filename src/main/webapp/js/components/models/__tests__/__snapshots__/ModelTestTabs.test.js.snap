// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModelTestTabs /> Snapshots renders correctly when a batch test tab currently active 1`] = `
<div
  id="ModelTest"
>
  <Deprecated
    fluid={true}
  >
    <Deprecated>
      <div
        id="ReadProject"
      >
        <withTheme(Tabs)
          align="left"
          onTabSelected={[Function]}
          selectedIndex={1}
          tabPanels={
            Array [
              <Connect(UtteranceTestContainer)
                clearRunSingleUtterance={[Function]}
                client={
                  Object {
                    "id": "2",
                  }
                }
                csrfToken="123"
                digitalModelTestResults={Object {}}
                dispatch={[Function]}
                history={Object {}}
                match={
                  Object {
                    "path": "/models/batchtest",
                  }
                }
                model={
                  Object {
                    "modelToken": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                  }
                }
                modelId="3"
                modelTestResults={
                  Object {
                    "evaluations": Array [
                      Object {
                        "entities": Array [
                          Object {
                            "name": "date",
                            "value": "june 13",
                          },
                        ],
                        "intents": Array [
                          Object {
                            "intent": "reservation-query",
                            "score": 0.49544276870123316,
                          },
                          Object {
                            "intent": "agent-query",
                            "score": 0.42004871402243316,
                          },
                          Object {
                            "intent": "None_None",
                            "score": 0.14285217997731586,
                          },
                        ],
                        "transformations": Array [
                          Object {
                            "id": "rx-mapper.non-breaking-space-regex",
                            "result": "i want to cancel for june 13",
                          },
                          Object {
                            "id": "wsp",
                            "result": "i want to cancel for june 13",
                          },
                        ],
                        "utterance": "i want to cancel for june 13",
                      },
                    ],
                    "modelId": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                    "projectId": "7",
                    "status": "SUCCESS",
                    "type": "UTTERANCES",
                  }
                }
                projectId="7"
                runSingleUtterance=""
                speechModelTestResults={Object {}}
                userId="abc"
              />,
              <Connect(BatchTestContainer)
                client={
                  Object {
                    "id": "2",
                  }
                }
                csrfToken="123"
                digitalModelTestResults={Object {}}
                dispatch={[Function]}
                history={Object {}}
                match={
                  Object {
                    "path": "/models/batchtest",
                  }
                }
                model={
                  Object {
                    "modelToken": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                  }
                }
                modelId="3"
                modelTestResults={
                  Object {
                    "evaluations": Array [
                      Object {
                        "entities": Array [
                          Object {
                            "name": "date",
                            "value": "june 13",
                          },
                        ],
                        "intents": Array [
                          Object {
                            "intent": "reservation-query",
                            "score": 0.49544276870123316,
                          },
                          Object {
                            "intent": "agent-query",
                            "score": 0.42004871402243316,
                          },
                          Object {
                            "intent": "None_None",
                            "score": 0.14285217997731586,
                          },
                        ],
                        "transformations": Array [
                          Object {
                            "id": "rx-mapper.non-breaking-space-regex",
                            "result": "i want to cancel for june 13",
                          },
                          Object {
                            "id": "wsp",
                            "result": "i want to cancel for june 13",
                          },
                        ],
                        "utterance": "i want to cancel for june 13",
                      },
                    ],
                    "modelId": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                    "projectId": "7",
                    "status": "SUCCESS",
                    "type": "UTTERANCES",
                  }
                }
                onReviewDatasetsClick={[Function]}
                onRunSingleUtterance={[Function]}
                onTuneModelClick={[Function]}
                projectId="7"
                speechModelTestResults={Object {}}
                userId="abc"
              />,
            ]
          }
          tabs={
            Array [
              "Utterance Test",
              "Batch Test",
            ]
          }
        />
      </div>
    </Deprecated>
  </Deprecated>
</div>
`;

exports[`<ModelTestTabs /> Snapshots renders correctly when a utterance tab currently active 1`] = `
<div
  id="ModelTest"
>
  <Deprecated
    fluid={true}
  >
    <Deprecated>
      <div
        id="ReadProject"
      >
        <withTheme(Tabs)
          align="left"
          onTabSelected={[Function]}
          selectedIndex={0}
          tabPanels={
            Array [
              <Connect(UtteranceTestContainer)
                clearRunSingleUtterance={[Function]}
                client={
                  Object {
                    "id": "2",
                  }
                }
                csrfToken="123"
                digitalModelTestResults={Object {}}
                dispatch={[Function]}
                history={Object {}}
                match={
                  Object {
                    "path": "/models/test",
                  }
                }
                model={
                  Object {
                    "modelToken": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                  }
                }
                modelId="3"
                modelTestResults={
                  Object {
                    "evaluations": Array [
                      Object {
                        "entities": Array [
                          Object {
                            "name": "date",
                            "value": "june 13",
                          },
                        ],
                        "intents": Array [
                          Object {
                            "intent": "reservation-query",
                            "score": 0.49544276870123316,
                          },
                          Object {
                            "intent": "agent-query",
                            "score": 0.42004871402243316,
                          },
                          Object {
                            "intent": "None_None",
                            "score": 0.14285217997731586,
                          },
                        ],
                        "transformations": Array [
                          Object {
                            "id": "rx-mapper.non-breaking-space-regex",
                            "result": "i want to cancel for june 13",
                          },
                          Object {
                            "id": "wsp",
                            "result": "i want to cancel for june 13",
                          },
                        ],
                        "utterance": "i want to cancel for june 13",
                      },
                    ],
                    "modelId": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                    "projectId": "7",
                    "status": "SUCCESS",
                    "type": "UTTERANCES",
                  }
                }
                projectId="7"
                runSingleUtterance=""
                speechModelTestResults={Object {}}
                userId="abc"
              />,
              <Connect(BatchTestContainer)
                client={
                  Object {
                    "id": "2",
                  }
                }
                csrfToken="123"
                digitalModelTestResults={Object {}}
                dispatch={[Function]}
                history={Object {}}
                match={
                  Object {
                    "path": "/models/test",
                  }
                }
                model={
                  Object {
                    "modelToken": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                  }
                }
                modelId="3"
                modelTestResults={
                  Object {
                    "evaluations": Array [
                      Object {
                        "entities": Array [
                          Object {
                            "name": "date",
                            "value": "june 13",
                          },
                        ],
                        "intents": Array [
                          Object {
                            "intent": "reservation-query",
                            "score": 0.49544276870123316,
                          },
                          Object {
                            "intent": "agent-query",
                            "score": 0.42004871402243316,
                          },
                          Object {
                            "intent": "None_None",
                            "score": 0.14285217997731586,
                          },
                        ],
                        "transformations": Array [
                          Object {
                            "id": "rx-mapper.non-breaking-space-regex",
                            "result": "i want to cancel for june 13",
                          },
                          Object {
                            "id": "wsp",
                            "result": "i want to cancel for june 13",
                          },
                        ],
                        "utterance": "i want to cancel for june 13",
                      },
                    ],
                    "modelId": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                    "projectId": "7",
                    "status": "SUCCESS",
                    "type": "UTTERANCES",
                  }
                }
                onReviewDatasetsClick={[Function]}
                onRunSingleUtterance={[Function]}
                onTuneModelClick={[Function]}
                projectId="7"
                speechModelTestResults={Object {}}
                userId="abc"
              />,
            ]
          }
          tabs={
            Array [
              "Utterance Test",
              "Batch Test",
            ]
          }
        />
      </div>
    </Deprecated>
  </Deprecated>
</div>
`;

exports[`<ModelTestTabs /> Snapshots renders correctly with two tabs 1`] = `
<div
  id="ModelTest"
>
  <Deprecated
    fluid={true}
  >
    <Deprecated>
      <div
        id="ReadProject"
      >
        <withTheme(Tabs)
          align="left"
          onTabSelected={[Function]}
          selectedIndex={0}
          tabPanels={
            Array [
              <Connect(UtteranceTestContainer)
                clearRunSingleUtterance={[Function]}
                client={
                  Object {
                    "id": "2",
                  }
                }
                csrfToken="123"
                digitalModelTestResults={Object {}}
                dispatch={[Function]}
                history={Object {}}
                match={
                  Object {
                    "path": "/models/test",
                  }
                }
                model={
                  Object {
                    "modelToken": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                  }
                }
                modelId="3"
                modelTestResults={
                  Object {
                    "evaluations": Array [
                      Object {
                        "entities": Array [
                          Object {
                            "name": "date",
                            "value": "june 13",
                          },
                        ],
                        "intents": Array [
                          Object {
                            "intent": "reservation-query",
                            "score": 0.49544276870123316,
                          },
                          Object {
                            "intent": "agent-query",
                            "score": 0.42004871402243316,
                          },
                          Object {
                            "intent": "None_None",
                            "score": 0.14285217997731586,
                          },
                        ],
                        "transformations": Array [
                          Object {
                            "id": "rx-mapper.non-breaking-space-regex",
                            "result": "i want to cancel for june 13",
                          },
                          Object {
                            "id": "wsp",
                            "result": "i want to cancel for june 13",
                          },
                        ],
                        "utterance": "i want to cancel for june 13",
                      },
                    ],
                    "modelId": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                    "projectId": "7",
                    "status": "SUCCESS",
                    "type": "UTTERANCES",
                  }
                }
                projectId="7"
                runSingleUtterance=""
                speechModelTestResults={Object {}}
                userId="abc"
              />,
              <Connect(BatchTestContainer)
                client={
                  Object {
                    "id": "2",
                  }
                }
                csrfToken="123"
                digitalModelTestResults={Object {}}
                dispatch={[Function]}
                history={Object {}}
                match={
                  Object {
                    "path": "/models/test",
                  }
                }
                model={
                  Object {
                    "modelToken": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                  }
                }
                modelId="3"
                modelTestResults={
                  Object {
                    "evaluations": Array [
                      Object {
                        "entities": Array [
                          Object {
                            "name": "date",
                            "value": "june 13",
                          },
                        ],
                        "intents": Array [
                          Object {
                            "intent": "reservation-query",
                            "score": 0.49544276870123316,
                          },
                          Object {
                            "intent": "agent-query",
                            "score": 0.42004871402243316,
                          },
                          Object {
                            "intent": "None_None",
                            "score": 0.14285217997731586,
                          },
                        ],
                        "transformations": Array [
                          Object {
                            "id": "rx-mapper.non-breaking-space-regex",
                            "result": "i want to cancel for june 13",
                          },
                          Object {
                            "id": "wsp",
                            "result": "i want to cancel for june 13",
                          },
                        ],
                        "utterance": "i want to cancel for june 13",
                      },
                    ],
                    "modelId": "b61693f6-fa65-4308-af09-b32872dfb5f6",
                    "projectId": "7",
                    "status": "SUCCESS",
                    "type": "UTTERANCES",
                  }
                }
                onReviewDatasetsClick={[Function]}
                onRunSingleUtterance={[Function]}
                onTuneModelClick={[Function]}
                projectId="7"
                speechModelTestResults={Object {}}
                userId="abc"
              />,
            ]
          }
          tabs={
            Array [
              "Utterance Test",
              "Batch Test",
            ]
          }
        />
      </div>
    </Deprecated>
  </Deprecated>
</div>
`;
